src/packages/mikroorm/src/introspection/files/schema-entity-file.ts

```diff
@@ -0,0 +1,262 @@
+import type {
+	Dictionary,
+	EntityMetadata,
+	EntityProperty,
+	NamingStrategy,
+	Platform,
+} from '@mikro-orm/core';
+import { ReferenceType, Utils } from '@mikro-orm/core';
+
+import { BaseFile } from './base-file';
+
+export class SchemaEntityFile extends BaseFile {
+	protected readonly coreImports = new Set<string>();
+	protected readonly scalarImports = new Set<string>();
+	protected readonly entityImports = new Set<string>();
+	protected readonly enumImports = new Set<string>();
+
+	constructor(
+		protected readonly meta: EntityMetadata,
+		protected readonly namingStrategy: NamingStrategy,
+		protected readonly platform: Platform
+	) {
+		super(meta, namingStrategy, platform);
+	}
+
+	getBasePath() {
+		const dirName = this.pascalToKebabCaseString(this.meta.className);
+		return `backend/schema/${dirName}/`;
+	}
+
+	getBaseName() {
+		return 'entity.ts';
+	}
+
+	generate(): string {
+		const enumDefinitions: string[] = [];
+		let classBody = '\n';
+		const props = Object.values(this.meta.properties);
+		props.forEach((prop) => {
+			const decorator = this.getPropertyDecorator(prop);
+			const definition = this.getPropertyDefinition(prop);
+
+			if (!classBody.endsWith('\n\n')) {
+				classBody += '\n';
+			}
+
+			// Add a summary field if we have a name or title attribute
+			if (['name', 'title'].includes(prop.name.toLowerCase())) {
+				this.coreImports.add('SummaryField');
+				classBody += `\t@SummaryField()\n`;
+			}
+			classBody += decorator;
+			classBody += definition;
+
+			if (props[props.length - 1] !== prop) classBody += '\n';
+
+			if (prop.enum) {
+				const enumClassName = this.namingStrategy.getClassName(
+					this.meta.collection + '_' + prop.fieldNames[0],
+					'_'
+				);
+				enumDefinitions.push(this.getEnumClassDefinition(enumClassName));
+			}
+		});
+
+		let ret = '';
```

Comment:
What does ret mean?

Reply:
supposed to be the `ret`urned string but I dont like it now 😄 have renamed them all to `file`